{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneName</th>\n",
       "      <th>AccNum</th>\n",
       "      <th>Chrom</th>\n",
       "      <th>mRNALength</th>\n",
       "      <th>ORFLength</th>\n",
       "      <th>fpUTR_length</th>\n",
       "      <th>tpUTR_length</th>\n",
       "      <th>sumIntronLengths</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UBE2Q1</td>\n",
       "      <td>NM_017582</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3223</td>\n",
       "      <td>1269</td>\n",
       "      <td>91</td>\n",
       "      <td>1863</td>\n",
       "      <td>6847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DUOXA1</td>\n",
       "      <td>NM_001276267</td>\n",
       "      <td>chr15</td>\n",
       "      <td>3032</td>\n",
       "      <td>897</td>\n",
       "      <td>395</td>\n",
       "      <td>1740</td>\n",
       "      <td>9479</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneName        AccNum  Chrom  mRNALength  ORFLength  fpUTR_length  \\\n",
       "0   UBE2Q1     NM_017582   chr1        3223       1269            91   \n",
       "1   DUOXA1  NM_001276267  chr15        3032        897           395   \n",
       "\n",
       "   tpUTR_length  sumIntronLengths  \n",
       "0          1863              6847  \n",
       "1          1740              9479  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filteredGeneList = pd.read_csv('../RBC/Database/filteredGenesDetails_human_240118.txt',sep='\\t')\n",
    "acc2gene={}\n",
    "gene2acc={}\n",
    "for idx,row in filteredGeneList.iterrows():\n",
    "    acc2gene[row['AccNum']] = row['GeneName']\n",
    "    gene2acc[row['GeneName']] = row['AccNum']\n",
    "filteredGeneList.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AccNum</th>\n",
       "      <th>TIS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NM_017582</td>\n",
       "      <td>101.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NM_001276267</td>\n",
       "      <td>111.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NM_173469</td>\n",
       "      <td>79.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NM_001330474</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NM_014372</td>\n",
       "      <td>102.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         AccNum    TIS\n",
       "0     NM_017582  101.0\n",
       "1  NM_001276267  111.0\n",
       "2     NM_173469   79.0\n",
       "3  NM_001330474   34.0\n",
       "4     NM_014372  102.0"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TIS = pd.read_csv('../RBC/Database/accnum2tis.csv')\n",
    "TIS.columns = ['AccNum','TIS']\n",
    "TIS.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>logLikelihood</th>\n",
       "      <th>Gene</th>\n",
       "      <th>AccNum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>-14.542486</td>\n",
       "      <td>MYO1D</td>\n",
       "      <td>NM_015194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>-14.550683</td>\n",
       "      <td>PPP1R14B</td>\n",
       "      <td>NM_138689</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  logLikelihood      Gene     AccNum\n",
       "0           0     -14.542486     MYO1D  NM_015194\n",
       "1           1     -14.550683  PPP1R14B  NM_138689"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "logL_L24 = pd.read_csv('../RBC/MOTIF/SeqLogo/gene_likelihood/ER_L24_t1_enriched_likelihood.csv')\n",
    "logL_S15_t0 = pd.read_csv('../RBC/MOTIF/SeqLogo/gene_likelihood/ER_S15_t0_enriched_likelihood.csv')\n",
    "logL_S15_t1 = pd.read_csv('../RBC/MOTIF/SeqLogo/gene_likelihood/ER_S15_t1_enriched_likelihood.csv')\n",
    "logL_S15_t2 = pd.read_csv('../RBC/MOTIF/SeqLogo/gene_likelihood/ER_S15_t2_enriched_likelihood.csv')\n",
    "logL_L24.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "mfe_fp = pd.read_csv('../RBC/MOTIF/rnafold/fpUTR_minE_normbyLen_fillmean.csv')\n",
    "mfe_fp_orf = pd.read_csv('../RBC/MOTIF/rnafold/utr_and_orf_minE_normbyLen.csv')\n",
    "mfe_fp.columns = ['AccNum','MFE_fpUTR_normed']\n",
    "mfe_fp_orf.columns = ['AccNum','MFE_fpUTRandORF_normed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7890\n"
     ]
    }
   ],
   "source": [
    "comb = pd.DataFrame(logL_L24.loc[:,'Gene'])\n",
    "print(len(comb))\n",
    "comb['AccNum'] = logL_L24['AccNum']\n",
    "comb['seqlogo_logL_L24_t1'] = logL_L24['logLikelihood']\n",
    "comb['seqlogo_logL_S15_t0'] = logL_S15_t0['logLikelihood']\n",
    "comb['seqlogo_logL_S15_t1'] = logL_S15_t1['logLikelihood']\n",
    "comb['seqlogo_logL_S15_t2'] = logL_S15_t2['logLikelihood']\n",
    "comb = pd.merge(comb, TIS, how='inner', on=['AccNum'])\n",
    "comb = pd.merge(comb, mfe_fp, how='inner', on=['AccNum'])\n",
    "comb = pd.merge(comb, mfe_fp_orf, how='inner', on=['AccNum'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>AccNum</th>\n",
       "      <th>seqlogo_logL_L24_t1</th>\n",
       "      <th>seqlogo_logL_S15_t0</th>\n",
       "      <th>seqlogo_logL_S15_t1</th>\n",
       "      <th>seqlogo_logL_S15_t2</th>\n",
       "      <th>TIS</th>\n",
       "      <th>MFE_fpUTR_normed</th>\n",
       "      <th>MFE_fpUTRandORF_normed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO1D</td>\n",
       "      <td>NM_015194</td>\n",
       "      <td>-14.542486</td>\n",
       "      <td>-15.117326</td>\n",
       "      <td>-15.447782</td>\n",
       "      <td>-13.241358</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.522658</td>\n",
       "      <td>-0.504090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP1R14B</td>\n",
       "      <td>NM_138689</td>\n",
       "      <td>-14.550683</td>\n",
       "      <td>-15.397989</td>\n",
       "      <td>-15.667537</td>\n",
       "      <td>-13.566780</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.644104</td>\n",
       "      <td>-0.622038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGFG1</td>\n",
       "      <td>NM_001135187</td>\n",
       "      <td>-14.663012</td>\n",
       "      <td>-15.412761</td>\n",
       "      <td>-15.725220</td>\n",
       "      <td>-13.634400</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.666440</td>\n",
       "      <td>-0.598229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LAMP1</td>\n",
       "      <td>NM_005561</td>\n",
       "      <td>-14.668466</td>\n",
       "      <td>-15.438871</td>\n",
       "      <td>-15.744788</td>\n",
       "      <td>-13.810126</td>\n",
       "      <td>98.0</td>\n",
       "      <td>-0.519715</td>\n",
       "      <td>-0.509029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFG3L2</td>\n",
       "      <td>NM_006796</td>\n",
       "      <td>-14.691183</td>\n",
       "      <td>-15.474773</td>\n",
       "      <td>-15.753664</td>\n",
       "      <td>-13.876347</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.560619</td>\n",
       "      <td>-0.620646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7885</th>\n",
       "      <td>SDCBP</td>\n",
       "      <td>NM_001348341</td>\n",
       "      <td>-33.683685</td>\n",
       "      <td>-29.864479</td>\n",
       "      <td>-28.686124</td>\n",
       "      <td>-32.418184</td>\n",
       "      <td>44.0</td>\n",
       "      <td>-0.312993</td>\n",
       "      <td>-0.308519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7886</th>\n",
       "      <td>RPRD1A</td>\n",
       "      <td>NM_001303411</td>\n",
       "      <td>-33.846032</td>\n",
       "      <td>-29.997533</td>\n",
       "      <td>-28.854125</td>\n",
       "      <td>-32.556170</td>\n",
       "      <td>79.0</td>\n",
       "      <td>-0.464314</td>\n",
       "      <td>-0.414923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7887</th>\n",
       "      <td>FASTKD1</td>\n",
       "      <td>NM_001322046</td>\n",
       "      <td>-33.846032</td>\n",
       "      <td>-30.293971</td>\n",
       "      <td>-29.278236</td>\n",
       "      <td>-33.208495</td>\n",
       "      <td>73.0</td>\n",
       "      <td>-0.330131</td>\n",
       "      <td>-0.329384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7888</th>\n",
       "      <td>SCRN3</td>\n",
       "      <td>NM_024583</td>\n",
       "      <td>-33.975719</td>\n",
       "      <td>-31.335351</td>\n",
       "      <td>-29.994056</td>\n",
       "      <td>-33.585789</td>\n",
       "      <td>67.0</td>\n",
       "      <td>-0.305612</td>\n",
       "      <td>-0.258283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7889</th>\n",
       "      <td>DMXL1</td>\n",
       "      <td>NM_001290322</td>\n",
       "      <td>-34.936676</td>\n",
       "      <td>-32.137272</td>\n",
       "      <td>-30.459612</td>\n",
       "      <td>-36.070696</td>\n",
       "      <td>36.0</td>\n",
       "      <td>-0.328342</td>\n",
       "      <td>-0.325718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7890 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene        AccNum  seqlogo_logL_L24_t1  seqlogo_logL_S15_t0  \\\n",
       "0        MYO1D     NM_015194           -14.542486           -15.117326   \n",
       "1     PPP1R14B     NM_138689           -14.550683           -15.397989   \n",
       "2        AGFG1  NM_001135187           -14.663012           -15.412761   \n",
       "3        LAMP1     NM_005561           -14.668466           -15.438871   \n",
       "4       AFG3L2     NM_006796           -14.691183           -15.474773   \n",
       "...        ...           ...                  ...                  ...   \n",
       "7885     SDCBP  NM_001348341           -33.683685           -29.864479   \n",
       "7886    RPRD1A  NM_001303411           -33.846032           -29.997533   \n",
       "7887   FASTKD1  NM_001322046           -33.846032           -30.293971   \n",
       "7888     SCRN3     NM_024583           -33.975719           -31.335351   \n",
       "7889     DMXL1  NM_001290322           -34.936676           -32.137272   \n",
       "\n",
       "      seqlogo_logL_S15_t1  seqlogo_logL_S15_t2    TIS  MFE_fpUTR_normed  \\\n",
       "0              -15.447782           -13.241358   91.0         -0.522658   \n",
       "1              -15.667537           -13.566780   91.0         -0.644104   \n",
       "2              -15.725220           -13.634400  113.0         -0.666440   \n",
       "3              -15.744788           -13.810126   98.0         -0.519715   \n",
       "4              -15.753664           -13.876347  113.0         -0.560619   \n",
       "...                   ...                  ...    ...               ...   \n",
       "7885           -28.686124           -32.418184   44.0         -0.312993   \n",
       "7886           -28.854125           -32.556170   79.0         -0.464314   \n",
       "7887           -29.278236           -33.208495   73.0         -0.330131   \n",
       "7888           -29.994056           -33.585789   67.0         -0.305612   \n",
       "7889           -30.459612           -36.070696   36.0         -0.328342   \n",
       "\n",
       "      MFE_fpUTRandORF_normed  \n",
       "0                  -0.504090  \n",
       "1                  -0.622038  \n",
       "2                  -0.598229  \n",
       "3                  -0.509029  \n",
       "4                  -0.620646  \n",
       "...                      ...  \n",
       "7885               -0.308519  \n",
       "7886               -0.414923  \n",
       "7887               -0.329384  \n",
       "7888               -0.258283  \n",
       "7889               -0.325718  \n",
       "\n",
       "[7890 rows x 9 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#comb.to_csv('collate_genes.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "ER_files = glob.glob('../RBC/Deseq2 Analysis/shift15/coding/csv_raw/ER*.csv')\n",
    "deseq=pd.read_csv(ER_files[0])\n",
    "df=pd.DataFrame({'AccNum':deseq['Unnamed: 0']})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../RBC/Deseq2 Analysis/shift15/coding/csv_raw\\\\ER_L24_t1.csv',\n",
       " '../RBC/Deseq2 Analysis/shift15/coding/csv_raw\\\\ER_L24_t2.csv',\n",
       " '../RBC/Deseq2 Analysis/shift15/coding/csv_raw\\\\ER_S15_t0.csv',\n",
       " '../RBC/Deseq2 Analysis/shift15/coding/csv_raw\\\\ER_S15_t1.csv',\n",
       " '../RBC/Deseq2 Analysis/shift15/coding/csv_raw\\\\ER_S15_t2.csv']"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ER_files[1:6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "for ER_file in ER_files[1:6]:\n",
    "    keyword = ER_file.split('\\\\')[-1][:-4]\n",
    "    if keyword == 'ER_L24_t2': continue\n",
    "    deseq = pd.read_csv(ER_file)\n",
    "    df[keyword+'_logFC'] = deseq['log2FoldChange']\n",
    "    df[keyword+'_change'] = 'No'\n",
    "    df.loc[df[keyword+'_logFC']>1,keyword+'_change'] = 'enrich'\n",
    "    df.loc[df[keyword+'_logFC']<-1,keyword+'_change'] = 'deplete'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb=pd.merge(comb, df, how='right', on=['AccNum'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>AccNum</th>\n",
       "      <th>seqlogo_logL_L24_t1</th>\n",
       "      <th>seqlogo_logL_S15_t0</th>\n",
       "      <th>seqlogo_logL_S15_t1</th>\n",
       "      <th>seqlogo_logL_S15_t2</th>\n",
       "      <th>TIS</th>\n",
       "      <th>MFE_fpUTR_normed</th>\n",
       "      <th>MFE_fpUTRandORF_normed</th>\n",
       "      <th>ER_L24_t1_logFC</th>\n",
       "      <th>ER_L24_t1_change</th>\n",
       "      <th>ER_S15_t0_logFC</th>\n",
       "      <th>ER_S15_t0_change</th>\n",
       "      <th>ER_S15_t1_logFC</th>\n",
       "      <th>ER_S15_t1_change</th>\n",
       "      <th>ER_S15_t2_logFC</th>\n",
       "      <th>ER_S15_t2_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO1D</td>\n",
       "      <td>NM_015194</td>\n",
       "      <td>-14.542486</td>\n",
       "      <td>-15.117326</td>\n",
       "      <td>-15.447782</td>\n",
       "      <td>-13.241358</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.522658</td>\n",
       "      <td>-0.504090</td>\n",
       "      <td>-0.489914</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.303135</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.466139</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.485069</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP1R14B</td>\n",
       "      <td>NM_138689</td>\n",
       "      <td>-14.550683</td>\n",
       "      <td>-15.397989</td>\n",
       "      <td>-15.667537</td>\n",
       "      <td>-13.566780</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.644104</td>\n",
       "      <td>-0.622038</td>\n",
       "      <td>1.073262</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.095194</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.694895</td>\n",
       "      <td>enrich</td>\n",
       "      <td>0.655761</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGFG1</td>\n",
       "      <td>NM_001135187</td>\n",
       "      <td>-14.663012</td>\n",
       "      <td>-15.412761</td>\n",
       "      <td>-15.725220</td>\n",
       "      <td>-13.634400</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.666440</td>\n",
       "      <td>-0.598229</td>\n",
       "      <td>-0.198402</td>\n",
       "      <td>No</td>\n",
       "      <td>0.158027</td>\n",
       "      <td>No</td>\n",
       "      <td>0.374909</td>\n",
       "      <td>No</td>\n",
       "      <td>0.282730</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LAMP1</td>\n",
       "      <td>NM_005561</td>\n",
       "      <td>-14.668466</td>\n",
       "      <td>-15.438871</td>\n",
       "      <td>-15.744788</td>\n",
       "      <td>-13.810126</td>\n",
       "      <td>98.0</td>\n",
       "      <td>-0.519715</td>\n",
       "      <td>-0.509029</td>\n",
       "      <td>0.290764</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.145868</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.270792</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.268401</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFG3L2</td>\n",
       "      <td>NM_006796</td>\n",
       "      <td>-14.691183</td>\n",
       "      <td>-15.474773</td>\n",
       "      <td>-15.753664</td>\n",
       "      <td>-13.876347</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.560619</td>\n",
       "      <td>-0.620646</td>\n",
       "      <td>-0.085239</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.304327</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.431552</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.262779</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7985</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001080398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312778</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.140486</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.174372</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.250475</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001170330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.072176</td>\n",
       "      <td>No</td>\n",
       "      <td>1.260953</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.604759</td>\n",
       "      <td>enrich</td>\n",
       "      <td>0.775510</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7987</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001308632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.179873</td>\n",
       "      <td>No</td>\n",
       "      <td>0.280138</td>\n",
       "      <td>No</td>\n",
       "      <td>0.161024</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.197482</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7988</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001105570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.242662</td>\n",
       "      <td>No</td>\n",
       "      <td>0.582128</td>\n",
       "      <td>No</td>\n",
       "      <td>0.184474</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.056924</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7989</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_014727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.392056</td>\n",
       "      <td>No</td>\n",
       "      <td>0.392558</td>\n",
       "      <td>No</td>\n",
       "      <td>0.385641</td>\n",
       "      <td>No</td>\n",
       "      <td>0.339236</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7990 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene        AccNum  seqlogo_logL_L24_t1  seqlogo_logL_S15_t0  \\\n",
       "0        MYO1D     NM_015194           -14.542486           -15.117326   \n",
       "1     PPP1R14B     NM_138689           -14.550683           -15.397989   \n",
       "2        AGFG1  NM_001135187           -14.663012           -15.412761   \n",
       "3        LAMP1     NM_005561           -14.668466           -15.438871   \n",
       "4       AFG3L2     NM_006796           -14.691183           -15.474773   \n",
       "...        ...           ...                  ...                  ...   \n",
       "7985       NaN  NM_001080398                  NaN                  NaN   \n",
       "7986       NaN  NM_001170330                  NaN                  NaN   \n",
       "7987       NaN  NM_001308632                  NaN                  NaN   \n",
       "7988       NaN  NM_001105570                  NaN                  NaN   \n",
       "7989       NaN     NM_014727                  NaN                  NaN   \n",
       "\n",
       "      seqlogo_logL_S15_t1  seqlogo_logL_S15_t2    TIS  MFE_fpUTR_normed  \\\n",
       "0              -15.447782           -13.241358   91.0         -0.522658   \n",
       "1              -15.667537           -13.566780   91.0         -0.644104   \n",
       "2              -15.725220           -13.634400  113.0         -0.666440   \n",
       "3              -15.744788           -13.810126   98.0         -0.519715   \n",
       "4              -15.753664           -13.876347  113.0         -0.560619   \n",
       "...                   ...                  ...    ...               ...   \n",
       "7985                  NaN                  NaN    NaN               NaN   \n",
       "7986                  NaN                  NaN    NaN               NaN   \n",
       "7987                  NaN                  NaN    NaN               NaN   \n",
       "7988                  NaN                  NaN    NaN               NaN   \n",
       "7989                  NaN                  NaN    NaN               NaN   \n",
       "\n",
       "      MFE_fpUTRandORF_normed  ER_L24_t1_logFC ER_L24_t1_change  \\\n",
       "0                  -0.504090        -0.489914               No   \n",
       "1                  -0.622038         1.073262           enrich   \n",
       "2                  -0.598229        -0.198402               No   \n",
       "3                  -0.509029         0.290764               No   \n",
       "4                  -0.620646        -0.085239               No   \n",
       "...                      ...              ...              ...   \n",
       "7985                     NaN        -0.312778               No   \n",
       "7986                     NaN         0.072176               No   \n",
       "7987                     NaN         0.179873               No   \n",
       "7988                     NaN         0.242662               No   \n",
       "7989                     NaN         0.392056               No   \n",
       "\n",
       "      ER_S15_t0_logFC ER_S15_t0_change  ER_S15_t1_logFC ER_S15_t1_change  \\\n",
       "0           -0.303135               No        -0.466139               No   \n",
       "1            1.095194           enrich         1.694895           enrich   \n",
       "2            0.158027               No         0.374909               No   \n",
       "3           -0.145868               No        -0.270792               No   \n",
       "4           -0.304327               No        -0.431552               No   \n",
       "...               ...              ...              ...              ...   \n",
       "7985        -0.140486               No        -0.174372               No   \n",
       "7986         1.260953           enrich         1.604759           enrich   \n",
       "7987         0.280138               No         0.161024               No   \n",
       "7988         0.582128               No         0.184474               No   \n",
       "7989         0.392558               No         0.385641               No   \n",
       "\n",
       "      ER_S15_t2_logFC ER_S15_t2_change  \n",
       "0           -0.485069               No  \n",
       "1            0.655761               No  \n",
       "2            0.282730               No  \n",
       "3           -0.268401               No  \n",
       "4           -0.262779               No  \n",
       "...               ...              ...  \n",
       "7985        -0.250475               No  \n",
       "7986         0.775510               No  \n",
       "7987        -0.197482               No  \n",
       "7988        -0.056924               No  \n",
       "7989         0.339236               No  \n",
       "\n",
       "[7990 rows x 17 columns]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'dict'>\n",
      "['NR_148357_chr1(+):11868-14361', 'NR_046018_chr1(+):11873-14408', 'NR_024540_chr1(-):14361-29369']\n"
     ]
    }
   ],
   "source": [
    "def get_fa(filename,onlyKeys = 'All'):\n",
    "    keyD = dict()\n",
    "    if onlyKeys!='All':\n",
    "        for k in onlyKeys: keyD[k] = None\n",
    "        \n",
    "    def addEntry(current,d):\n",
    "        entry = current.split('\\n')\n",
    "        if entry[0]!='' and (onlyKeys=='All' or keyD.has_key(entry[0])):\n",
    "            d[entry[0]] = ''.join(entry[1:])\n",
    "\n",
    "    d = dict()\n",
    "    bf = open(filename)\n",
    "    current = ['\\n']\n",
    "    for line in bf:\n",
    "        if line[0]=='>':\n",
    "            addEntry(''.join(current),d)\n",
    "            if len(line)>1: current = [line[1:]]\n",
    "            else: current = ['\\n']\n",
    "        else: current.append(line)\n",
    "    addEntry(''.join(current),d)\n",
    "    bf.close()\n",
    "    return d\n",
    "splicedGenes = get_fa('../RBC/Database/splicedGenes_240118.txt')\n",
    "print(type(splicedGenes))\n",
    "print(list(splicedGenes.keys())[0:3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'dict'>\n",
      "['NR_148357', 'NR_046018', 'NR_024540']\n"
     ]
    }
   ],
   "source": [
    "## truncate the keys to only accNum\n",
    "ks = list(splicedGenes.keys())\n",
    "for key in ks:\n",
    "    splicedGenes[key.split('_chr')[0]] = splicedGenes[key]\n",
    "    splicedGenes.pop(key)\n",
    "print(type(splicedGenes))\n",
    "print(list(splicedGenes.keys())[0:3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'dict'>\n",
      "['NM_017582', 'NM_001276267', 'NM_173469']\n"
     ]
    }
   ],
   "source": [
    "splicedORFs = get_fa('../RBC/Database/splicedORFs_240118.txt')\n",
    "print(type(splicedORFs))\n",
    "print(list(splicedORFs.keys())[0:3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "fpUTR={}\n",
    "tpUTR = {}\n",
    "ORFs = {}\n",
    "for idx,row in filteredGeneList.iterrows():\n",
    "    accNum = row['AccNum']\n",
    "    fpLen = row['fpUTR_length']\n",
    "    tpLen = row['tpUTR_length']\n",
    "    fpUTR[accNum] = splicedGenes[accNum][0:fpLen]\n",
    "    tpUTR[accNum] = splicedGenes[accNum][-tpLen:]\n",
    "    ORFs[accNum] = splicedORFs[accNum]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "def countGC(seq):\n",
    "    if len(seq)==0: return float('nan')\n",
    "    gc_all_percent = (seq.count('G')+seq.count('C'))/len(seq)\n",
    "    #codons = (re.findall('...?', seq))\n",
    "    #codon_tr=[''.join(s) for s in zip(*codons)]\n",
    "    #gc_last_nt = (codon_tr[2].count('G')+codon_tr[2].count('C'))/len(codons)\n",
    "    #return gc_all_percent#, \n",
    "    return gc_all_percent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "fpUTR_gc=[]\n",
    "for accNum in comb['AccNum']:\n",
    "    gc = countGC(fpUTR[accNum])\n",
    "    fpUTR_gc.append(gc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb['fpUTR_gc_percent'] = fpUTR_gc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb.to_csv('collate_genes_20may.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>AccNum</th>\n",
       "      <th>seqlogo_logL_L24_t1</th>\n",
       "      <th>seqlogo_logL_S15_t0</th>\n",
       "      <th>seqlogo_logL_S15_t1</th>\n",
       "      <th>seqlogo_logL_S15_t2</th>\n",
       "      <th>TIS</th>\n",
       "      <th>MFE_fpUTR_normed</th>\n",
       "      <th>MFE_fpUTRandORF_normed</th>\n",
       "      <th>ER_L24_t1_logFC</th>\n",
       "      <th>ER_L24_t1_change</th>\n",
       "      <th>ER_S15_t0_logFC</th>\n",
       "      <th>ER_S15_t0_change</th>\n",
       "      <th>ER_S15_t1_logFC</th>\n",
       "      <th>ER_S15_t1_change</th>\n",
       "      <th>ER_S15_t2_logFC</th>\n",
       "      <th>ER_S15_t2_change</th>\n",
       "      <th>fpUTR_gc_percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYO1D</td>\n",
       "      <td>NM_015194</td>\n",
       "      <td>-14.542486</td>\n",
       "      <td>-15.117326</td>\n",
       "      <td>-15.447782</td>\n",
       "      <td>-13.241358</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.522658</td>\n",
       "      <td>-0.504090</td>\n",
       "      <td>-0.489914</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.303135</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.466139</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.485069</td>\n",
       "      <td>No</td>\n",
       "      <td>0.807309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP1R14B</td>\n",
       "      <td>NM_138689</td>\n",
       "      <td>-14.550683</td>\n",
       "      <td>-15.397989</td>\n",
       "      <td>-15.667537</td>\n",
       "      <td>-13.566780</td>\n",
       "      <td>91.0</td>\n",
       "      <td>-0.644104</td>\n",
       "      <td>-0.622038</td>\n",
       "      <td>1.073262</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.095194</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.694895</td>\n",
       "      <td>enrich</td>\n",
       "      <td>0.655761</td>\n",
       "      <td>No</td>\n",
       "      <td>0.828358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AGFG1</td>\n",
       "      <td>NM_001135187</td>\n",
       "      <td>-14.663012</td>\n",
       "      <td>-15.412761</td>\n",
       "      <td>-15.725220</td>\n",
       "      <td>-13.634400</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.666440</td>\n",
       "      <td>-0.598229</td>\n",
       "      <td>-0.198402</td>\n",
       "      <td>No</td>\n",
       "      <td>0.158027</td>\n",
       "      <td>No</td>\n",
       "      <td>0.374909</td>\n",
       "      <td>No</td>\n",
       "      <td>0.282730</td>\n",
       "      <td>No</td>\n",
       "      <td>0.832000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LAMP1</td>\n",
       "      <td>NM_005561</td>\n",
       "      <td>-14.668466</td>\n",
       "      <td>-15.438871</td>\n",
       "      <td>-15.744788</td>\n",
       "      <td>-13.810126</td>\n",
       "      <td>98.0</td>\n",
       "      <td>-0.519715</td>\n",
       "      <td>-0.509029</td>\n",
       "      <td>0.290764</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.145868</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.270792</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.268401</td>\n",
       "      <td>No</td>\n",
       "      <td>0.786477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFG3L2</td>\n",
       "      <td>NM_006796</td>\n",
       "      <td>-14.691183</td>\n",
       "      <td>-15.474773</td>\n",
       "      <td>-15.753664</td>\n",
       "      <td>-13.876347</td>\n",
       "      <td>113.0</td>\n",
       "      <td>-0.560619</td>\n",
       "      <td>-0.620646</td>\n",
       "      <td>-0.085239</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.304327</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.431552</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.262779</td>\n",
       "      <td>No</td>\n",
       "      <td>0.798969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7985</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001080398</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.312778</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.140486</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.174372</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.250475</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7986</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001170330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.072176</td>\n",
       "      <td>No</td>\n",
       "      <td>1.260953</td>\n",
       "      <td>enrich</td>\n",
       "      <td>1.604759</td>\n",
       "      <td>enrich</td>\n",
       "      <td>0.775510</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7987</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001308632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.179873</td>\n",
       "      <td>No</td>\n",
       "      <td>0.280138</td>\n",
       "      <td>No</td>\n",
       "      <td>0.161024</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.197482</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7988</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_001105570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.242662</td>\n",
       "      <td>No</td>\n",
       "      <td>0.582128</td>\n",
       "      <td>No</td>\n",
       "      <td>0.184474</td>\n",
       "      <td>No</td>\n",
       "      <td>-0.056924</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7989</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_014727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.392056</td>\n",
       "      <td>No</td>\n",
       "      <td>0.392558</td>\n",
       "      <td>No</td>\n",
       "      <td>0.385641</td>\n",
       "      <td>No</td>\n",
       "      <td>0.339236</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7990 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene        AccNum  seqlogo_logL_L24_t1  seqlogo_logL_S15_t0  \\\n",
       "0        MYO1D     NM_015194           -14.542486           -15.117326   \n",
       "1     PPP1R14B     NM_138689           -14.550683           -15.397989   \n",
       "2        AGFG1  NM_001135187           -14.663012           -15.412761   \n",
       "3        LAMP1     NM_005561           -14.668466           -15.438871   \n",
       "4       AFG3L2     NM_006796           -14.691183           -15.474773   \n",
       "...        ...           ...                  ...                  ...   \n",
       "7985       NaN  NM_001080398                  NaN                  NaN   \n",
       "7986       NaN  NM_001170330                  NaN                  NaN   \n",
       "7987       NaN  NM_001308632                  NaN                  NaN   \n",
       "7988       NaN  NM_001105570                  NaN                  NaN   \n",
       "7989       NaN     NM_014727                  NaN                  NaN   \n",
       "\n",
       "      seqlogo_logL_S15_t1  seqlogo_logL_S15_t2    TIS  MFE_fpUTR_normed  \\\n",
       "0              -15.447782           -13.241358   91.0         -0.522658   \n",
       "1              -15.667537           -13.566780   91.0         -0.644104   \n",
       "2              -15.725220           -13.634400  113.0         -0.666440   \n",
       "3              -15.744788           -13.810126   98.0         -0.519715   \n",
       "4              -15.753664           -13.876347  113.0         -0.560619   \n",
       "...                   ...                  ...    ...               ...   \n",
       "7985                  NaN                  NaN    NaN               NaN   \n",
       "7986                  NaN                  NaN    NaN               NaN   \n",
       "7987                  NaN                  NaN    NaN               NaN   \n",
       "7988                  NaN                  NaN    NaN               NaN   \n",
       "7989                  NaN                  NaN    NaN               NaN   \n",
       "\n",
       "      MFE_fpUTRandORF_normed  ER_L24_t1_logFC ER_L24_t1_change  \\\n",
       "0                  -0.504090        -0.489914               No   \n",
       "1                  -0.622038         1.073262           enrich   \n",
       "2                  -0.598229        -0.198402               No   \n",
       "3                  -0.509029         0.290764               No   \n",
       "4                  -0.620646        -0.085239               No   \n",
       "...                      ...              ...              ...   \n",
       "7985                     NaN        -0.312778               No   \n",
       "7986                     NaN         0.072176               No   \n",
       "7987                     NaN         0.179873               No   \n",
       "7988                     NaN         0.242662               No   \n",
       "7989                     NaN         0.392056               No   \n",
       "\n",
       "      ER_S15_t0_logFC ER_S15_t0_change  ER_S15_t1_logFC ER_S15_t1_change  \\\n",
       "0           -0.303135               No        -0.466139               No   \n",
       "1            1.095194           enrich         1.694895           enrich   \n",
       "2            0.158027               No         0.374909               No   \n",
       "3           -0.145868               No        -0.270792               No   \n",
       "4           -0.304327               No        -0.431552               No   \n",
       "...               ...              ...              ...              ...   \n",
       "7985        -0.140486               No        -0.174372               No   \n",
       "7986         1.260953           enrich         1.604759           enrich   \n",
       "7987         0.280138               No         0.161024               No   \n",
       "7988         0.582128               No         0.184474               No   \n",
       "7989         0.392558               No         0.385641               No   \n",
       "\n",
       "      ER_S15_t2_logFC ER_S15_t2_change  fpUTR_gc_percent  \n",
       "0           -0.485069               No          0.807309  \n",
       "1            0.655761               No          0.828358  \n",
       "2            0.282730               No          0.832000  \n",
       "3           -0.268401               No          0.786477  \n",
       "4           -0.262779               No          0.798969  \n",
       "...               ...              ...               ...  \n",
       "7985        -0.250475               No               NaN  \n",
       "7986         0.775510               No               NaN  \n",
       "7987        -0.197482               No               NaN  \n",
       "7988        -0.056924               No               NaN  \n",
       "7989         0.339236               No               NaN  \n",
       "\n",
       "[7990 rows x 18 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fa(filename,onlyKeys = 'All'):\n",
    "    keyD = dict()\n",
    "    if onlyKeys!='All':\n",
    "        for k in onlyKeys: keyD[k] = None\n",
    "        \n",
    "    def addEntry(current,d):\n",
    "        entry = current.split('\\n')\n",
    "        if entry[0]!='' and (onlyKeys=='All' or keyD.has_key(entry[0])):\n",
    "            d[entry[0]] = ''.join(entry[1:])\n",
    "\n",
    "    d = dict()\n",
    "    bf = open(filename)\n",
    "    current = ['\\n']\n",
    "    for line in bf:\n",
    "        if line[0]=='>':\n",
    "            addEntry(''.join(current),d)\n",
    "            if len(line)>1: current = [line[1:]]\n",
    "            else: current = ['\\n']\n",
    "        else: current.append(line)\n",
    "    addEntry(''.join(current),d)\n",
    "    bf.close()\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "splicedGenes = get_fa('../RBC/Database/splicedGenes_240118.txt')\n",
    "## truncate the keys to only accNum\n",
    "ks = list(splicedGenes.keys())\n",
    "for key in ks:\n",
    "    splicedGenes[key.split('_chr')[0]] = splicedGenes[key]\n",
    "    splicedGenes.pop(key)\n",
    "splicedORFs = get_fa('../RBC/Database/splicedORFs_240118.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "fpUTR={}\n",
    "tpUTR = {}\n",
    "ORFs = {}\n",
    "for idx,row in filteredGeneList.iterrows():\n",
    "    accNum = row['AccNum']\n",
    "    fpLen = row['fpUTR_length']\n",
    "    tpLen = row['tpUTR_length']\n",
    "    fpUTR[accNum] = splicedGenes[accNum][0:fpLen]\n",
    "    tpUTR[accNum] = splicedGenes[accNum][-tpLen:]\n",
    "    ORFs[accNum] = splicedORFs[accNum]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "def gene2seq(acc):\n",
    "    fp=fpUTR[acc]\n",
    "    if len(fp)<8:\n",
    "        return 'NaN'\n",
    "    seq = fp[-8:]+ORFs[acc][:11]\n",
    "    #print('# of valid seqs: ',len(seqs))\n",
    "    return seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb['SeqLogo'] = [gene2seq(acc) for acc in comb['AccNum']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "comb.to_csv('collate_genes_with_seqlogo_seq.csv',index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
